TeamCode/src/main/java/t10/motion/hardware/PositionalMotor.java

```diff
@@ -4,52 +4,73 @@
 
 import java.security.InvalidParameterException;
 
+import t10.utils.PIDController;
+
+/**
+* A controller for DC motors that allows them to be controlled by setting a target position
+ */
 public class PositionalMotor {
     private final DcMotorEx motor;
-    private final double ticksPerRevolution;
-    private final double startPosition;
-    private final double endPosition;
-    private final double speed;
-    private double currentPosition;
-    private int coefficient;
-    private double targetPosition;
-    private double startTicksPosition;
-    private double positionBeforeMovement;
-
-    public PositionalMotor(DcMotorEx motor, double ticksPerRevolution, double startPosition, double endPosition, double speed, int coefficient) {
-        if (startPosition >= endPosition) {
-            throw new InvalidParameterException("startPosition must be less than endPosition");
+    private final PIDController pid;
+    private final float offsetTicks;
+    private final float ticksPerRotation;
+    private final float minBoundTicks;
+    private final float maxBoundTicks;
+    private final float speed;
+    private float currentPositionTicks = 0;
+    private float targetTicks;
+
+    /**
+     * Creates a PositionalMotor
+     * @param motor the motor to control
+     * @param ticksPerRotation the motor's ticks for one full revolution
+     * @param minBoundRotation the minimum inclusive rotation, in rotations, the motor will be allowed to rotate to (relative to initial position)
+     * @param maxBoundRotation the maximum inclusive rotation, in rotations, the motor will be allowed to rotate to (relative to initial position)
+     * @param initialRotation the initial position, in rotations, of the motor
+     * @param speed the speed for the motor to rotate at (can be negative for inverted rotation)
+     */
+    public PositionalMotor(DcMotorEx motor, float ticksPerRotation, float minBoundRotation, float maxBoundRotation, float initialRotation, float speed) {
+        if (minBoundRotation >= maxBoundRotation) {
+            throw new InvalidParameterException("minBoundRotation must be less than maxBoundRotation");
         }
 
         this.motor = motor;
-        this.ticksPerRevolution = ticksPerRevolution;
-        this.startPosition = startPosition;
-        this.currentPosition = startPosition;
-        this.endPosition = endPosition;
-        this.coefficient = coefficient;
+        this.ticksPerRotation = ticksPerRotation;
+        this.minBoundTicks = rotationToTicks(minBoundRotation);
+        this.maxBoundTicks = rotationToTicks(maxBoundRotation);
+        this.currentPositionTicks = rotationToTicks(initialRotation);
+        this.offsetTicks = rotationToTicks(initialRotation);
+
         this.speed = speed;
+
+        this.pid = new PIDController(0.1, 0, 0);
```

Comment:
kP, kI, kD must be set in the constructor because they are never the same for all motors and hardware